addons/slack/install.js

```diff
@@ -0,0 +1,56 @@
+/*
+This is the installation logic of the Lifecycle Manager. 
+*/
+
+const Fs = require('fs');
+const Sdk = require('@fusebit/add-on-sdk');
+const Superagent = require('superagent');
+
+const getTemplateFiles = (fileNames) =>
+  fileNames.reduce((a, c) => {
+    a[c] = Fs.readFileSync(__dirname + `/template/${c}`, { encoding: 'utf8' });
+    return a;
+  }, {});
+
+module.exports = async (ctx) => {
+  // Create the Add-On Handler
+  await Sdk.createFunction(ctx, {
+    enableStorage: true,
+    configurationSerialized: `# Add-on configuration settings
+${Object.keys(ctx.body.configuration)
+  .sort()
+  .map((k) => `${k}=${ctx.body.configuration[k]}`)
+  .join('\n')}
+`,
+    nodejs: {
+      files: getTemplateFiles(['index.js', 'package.json', 'slack.js']),
+    },
+    metadata: {
+      fusebit: {
+        editor: {
+          navigationPanel: {
+            hideFiles: ['index.js'],
+          },
+        },
+      },
+      ...ctx.body.metadata,
+    },
+  });
+
+  let registrationUrl = ctx.body.configuration.fusebit_events_registration;
+  let token = ctx.body.configuration.fusebit_events_token;
+  if (registrationUrl && token) {
+    Sdk.debug('Events API configuration was present');
+    await Superagent.post(registrationUrl)
+      .set('Authorization', `Bearer ${token}`)
+      .send({
+        app: ctx.body.configuration.slack_app_id,
+        team_id: ctx.body.configuration.slack_team_id,
+        handler: (await Sdk.getFunctionUrl(ctx)) + '/events/ingest',
+      });
```

Comment:
if registration fails, we should remove the function that was created.

Reply:
... actually, since this needs to handle the "upsert" case, we should remove the function only if it did not already exist when install was called.

Reply:
How do we know if it existed before given the PUT semantics of function creation? Is it worth adding another call to check the function existence before we try to create it? 

What's more problematic is that `Sdk.createStorage` is not reentrant currently, that will require more SDK work. Moreover if we want to similarly clean up storage only after we created it, we need new APIs to query if it exists. 

This feels like it is quickly ballooning into the work tracked by #732.

Reply:
Per discussion with Tomek and Benn, we'll leave this to fivequarters/q5#732, as we need to agree on the design and work and make `createStorage` reentrant.